app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageController.scala

```diff
@@ -314,39 +323,132 @@ class MultipleDisposalsTriageController @Inject() (
+                assetType                    = Some(complete.assetType),
+                taxYearAfter6April2020       = Some(taxYearAfter6April2020),
+                taxYear                      = taxYear
+              )
+          )
+        )
+    }
+
   def checkYourAnswers(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
     withMultipleDisposalTriageAnswers(request) {
       case (_, state, triageAnswers) =>
         val isIndividual = state.subscribedDetails.userType().isRight
 
         triageAnswers match {
-          case IncompleteMultipleDisposalsAnswers(None, _, _, _, _, _) if isIndividual =>
+          case IncompleteMultipleDisposalsAnswers(None, _, _, _, _, _, _, _) if isIndividual =>
             Redirect(routes.InitialTriageQuestionsController.whoIsIndividualRepresenting())
 
-          case IncompleteMultipleDisposalsAnswers(_, None, _, _, _, _) =>
+          case IncompleteMultipleDisposalsAnswers(_, None, _, _, _, _, _, _) =>
             Redirect(routes.MultipleDisposalsTriageController.guidance())
 
-          case IncompleteMultipleDisposalsAnswers(_, None, _, _, _, _) =>
+          case IncompleteMultipleDisposalsAnswers(_, None, _, _, _, _, _, _) =>
             Redirect(routes.MultipleDisposalsTriageController.howManyDisposals())
 
-          case IncompleteMultipleDisposalsAnswers(_, _, None, _, _, _) =>
+          case IncompleteMultipleDisposalsAnswers(_, _, None, _, _, _, _, _) =>
             Redirect(routes.MultipleDisposalsTriageController.wereYouAUKResident())
 
-          case IncompleteMultipleDisposalsAnswers(_, _, Some(true), _, None, _) =>
+          case IncompleteMultipleDisposalsAnswers(_, _, Some(true), _, None, _, _, _) =>
             Redirect(routes.MultipleDisposalsTriageController.wereAllPropertiesResidential())
 
-          case IncompleteMultipleDisposalsAnswers(_, _, Some(false), None, _, _) =>
+          case IncompleteMultipleDisposalsAnswers(_, _, Some(false), None, _, _, _, _) =>
             Redirect(routes.MultipleDisposalsTriageController.countryOfResidence())
 
-          case IncompleteMultipleDisposalsAnswers(_, _, Some(false), Some(country), _, _) =>
-            Ok(s"Country of residence: $country")
+          case IncompleteMultipleDisposalsAnswers(_, _, Some(false), Some(_), _, _, _, _) =>
+            Ok("Non-UK Residents not handled yet")
+
+          case IncompleteMultipleDisposalsAnswers(_, _, _, _, Some(false), _, _, _) =>
+            Ok("All are not residential properties")
+
+          case IncompleteMultipleDisposalsAnswers(_, _, _, _, _, _, None, _) =>
+            Redirect(routes.MultipleDisposalsTriageController.whenWereContractsExchanged())
 
-          case IncompleteMultipleDisposalsAnswers(_, _, _, _, Some(residentialStatus), _) =>
-            Ok(s"Were All Properties residential: $residentialStatus")
+          case IncompleteMultipleDisposalsAnswers(_, _, _, _, _, _, Some(true), _) =>
```

Comment:
we shouldn't allow
```scala
case IncompleteMultipleDisposalsAnswers(_, _, _, _, _, _, Some(true), None)
```
through. If we get that state somehow we should probably error out with a log message saying that no tax year was found when we expected one.